rust/automerge/src/parents.rs

```diff
@@ -10,23 +10,36 @@ pub struct Parents<'a> {
 
 impl<'a> Parents<'a> {
     pub fn path(&mut self) -> Vec<(ExId, Prop)> {
-        let mut path = self.collect::<Vec<_>>();
+        let mut path = self.map(|(id, prop, _)| (id, prop)).collect::<Vec<_>>();
         path.reverse();
         path
     }
+
+    pub fn visible_path(&mut self) -> Option<Vec<(ExId, Prop)>> {
+        let mut path = Vec::new();
+        for (id, prop, vis) in self {
+            if !vis {
+                return None;
+            }
+            path.push((id, prop))
+        }
+        path.reverse();
+        Some(path)
+    }
 }
 
 impl<'a> Iterator for Parents<'a> {
-    type Item = (ExId, Prop);
+    type Item = (ExId, Prop, bool);
```

Comment:
As per my comment on `OpSet::parent_object`, I think we should replace this three-tuple with a struct which names the fields to make it easier to read what the `bool` refers to.